[FSDP] No need for list() in _get_shard

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66958
* __->__ #66957
* #66956

chunk appears to return a tuple which is enough given that we just
index to the right chunk and discard the rest.

Differential Revision: [D31780799](https://our.internmc.facebook.com/intern/diff/D31780799/)

rohan-varma committed:
[FSDP] No need for list() in _get_shard

chunk appears to return a tuple which is enough given that we just
index to the right chunk and discard the rest.

Differential Revision: [D31780799](https://our.internmc.facebook.com/intern/diff/D31780799/)

[ghstack-poisoned]

zhaojuanmao reviewed (approved):
thanks! but could we land these changes after #66904 is landed, so that we have more tests to catch regression?

rohan-varma committed:
Update on "[FSDP] No need for list() in _get_shard"


chunk appears to return a tuple which is enough given that we just
index to the right chunk and discard the rest.

Differential Revision: [D31780799](https://our.internmc.facebook.com/intern/diff/D31780799/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[FSDP] No need for list() in _get_shard"


chunk appears to return a tuple which is enough given that we just
index to the right chunk and discard the rest.

Differential Revision: [D31780799](https://our.internmc.facebook.com/intern/diff/D31780799/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[FSDP] No need for list() in _get_shard"


chunk appears to return a tuple which is enough given that we just
index to the right chunk and discard the rest.

Differential Revision: [D31780799](https://our.internmc.facebook.com/intern/diff/D31780799/)

[ghstack-poisoned]
